From f27f3a18a9854098f5efc8e5472e21a4ffcdf0e6 Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Thu, 2 May 2013 18:08:42 +0200
Subject: [PATCH 403/482] 	* grub-core/tests/videotest_checksum.c
 (videotest_checksum): Error out 	if no unifont is found. 
 Restore original keyboard.

---
 ChangeLog                            | 6 ++++++
 grub-core/tests/videotest_checksum.c | 9 ++++++++-
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index de8acba..ef402f8 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,11 @@
 2013-05-02  Vladimir Serbinenko  <phcoder@gmail.com>
 
+	* grub-core/tests/videotest_checksum.c (videotest_checksum): Error out
+	if no unifont is found.
+	Restore original keyboard.
+
+2013-05-02  Vladimir Serbinenko  <phcoder@gmail.com>
+
 	* grub-core/loader/i386/linux.c (grub_linux_setup_video): Add
 	GRUB_VIDEO_ADAPTER_CAPTURE: to handled drived ids.
 
diff --git a/grub-core/tests/videotest_checksum.c b/grub-core/tests/videotest_checksum.c
index e57dae7..3c70f8c 100644
--- a/grub-core/tests/videotest_checksum.c
+++ b/grub-core/tests/videotest_checksum.c
@@ -265,7 +265,12 @@ static void
 videotest_checksum (void)
 {
   unsigned i;
-  grub_font_load ("unicode");
+  if (grub_font_load ("unicode") == 0)
+    {
+      grub_test_assert (0, "unicode font not found: %s", grub_errmsg);
+      return;
+    }
+  
   for (i = 0; i < ARRAY_SIZE (tests); i++)
     {
       grub_video_capture_start (&tests[i].mode_info,
@@ -277,6 +282,8 @@ videotest_checksum (void)
 
       char *args[] = { 0 };
       grub_command_execute ("videotest", 0, args);
+
+      grub_terminal_input_fake_sequence_end ();
       grub_video_checksum_end ();
       grub_video_capture_end ();
     }
-- 
1.8.2.1

